src/Application/Digital/Commands/ShareEntry.cs

```diff
@@ -0,0 +1,178 @@
+            {
+                throw new ConflictException("Expiry date cannot be in the past.");
+            }
+
+            var allowOperations = GenerateAllowOperations(request, entry.IsDirectory);
+            
+            await GrantPermission(entry, user, allowOperations, request.ExpiryDate, cancellationToken);
+
+            if (entry.IsDirectory)
+            {
+                // Update permissions for child Entries
+                var path = entry.Path.Equals("/") ? (entry.Path + entry.Name) : (entry.Path + "/" + entry.Name);
+                var childEntries = _context.Entries
+                    .Include(x => x.Uploader)
+                    .Include(x => x.Owner)
+                    .Where(x => x.Id != entry.Id 
+                                && x.Path.StartsWith(path)
+                                && x.OwnerId == entry.OwnerId)
+                    .ToList();
+                foreach (var childEntry in childEntries)
+                {
+                    var childAllowOperations = GenerateAllowOperations(request, childEntry.IsDirectory);
+                    await GrantPermission(childEntry, user, childAllowOperations, request.ExpiryDate, cancellationToken);
+                }
+            }
+
+            var result = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == request.EntryId
+                    && x.EmployeeId == request.UserId
+                , cancellationToken);
+            
+            await _context.SaveChangesAsync(cancellationToken);
+            return _mapper.Map<EntryPermissionDto>(result);
+        }
+
+        private async Task GrantPermission(
+            Entry entry,
+            User user,
+            string allowOperations,
+            DateTime expiryDate,
+            CancellationToken cancellationToken)
+        {
+            var existedPermission = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == entry.Id
+                    && x.EmployeeId == user.Id
+                , cancellationToken);
+            
+            if (existedPermission is null)
+            {
+                var entryPermission = new EntryPermission
+                {
+                    EmployeeId = user.Id,
+                    EntryId = entry.Id,
+                    AllowedOperations = allowOperations,
+                    ExpiryDateTime = LocalDateTime.FromDateTime(expiryDate),
+                    Employee = user,
+                    Entry = entry,
+                };
+                await _context.EntryPermissions.AddAsync(entryPermission, cancellationToken);
+            }
+            else
+            {
+                existedPermission.AllowedOperations = allowOperations;
+                existedPermission.ExpiryDateTime = LocalDateTime.FromDateTime(expiryDate);
+                _context.EntryPermissions.Update(existedPermission);
+            }
+            await _context.SaveChangesAsync(cancellationToken);
```

Comment:
why savechanges here?

Reply:
I forgot to remove that one after creating `GrantPermission` method.

Reply:
Fixed at https://github.com/ProFile-org/ProFile-BE/pull/336/commits/a07e593686e8415ecd98022b6448150d9d7ead78